cygnus-common/src/main/java/com/telefonica/iot/cygnus/metrics/CygnusMetrics.java

```diff
@@ -0,0 +1,379 @@
+            
+            if (thisMetrics == null) {
+                thisMetrics = new Metrics();
+                this.perSubserviceAggrMetrics.put(subservice, thisMetrics);
+            } // if
+            
+            Metrics otherMetrics = otherSuberviceMetrics.get(subservice);
+            thisMetrics.merge(otherMetrics);
+        } // for
+        
+        this.allAggrMetrics.merge(other.allAggrMetrics);
+    } // merge
+    
+    /**
+     * Gets the Json string for this metrics.
+     * @return The Json string for this metrics
+     */
+    public String toJsonString() {
+        String json = "{\"services\":{";
+        boolean firstService = true;
+        
+        for (String service : perServiceSubserviceMetrics.keySet()) {
+            if (firstService) {
+                json += "\"" + service + "\":{\"subservs\":{";
+                firstService = false;
+            } else {
+                json += ",\"" + service + "\":{\"subservs\":{";
+            } // if else
+            
+            HashMap<String, Metrics> subserviceMetrics = perServiceSubserviceMetrics.get(service);
+            boolean firstSubservice = true;
+            
+            for (String subservice : subserviceMetrics.keySet()) {
+                Metrics metrics = subserviceMetrics.get(subservice);
+                
+                if (firstSubservice) {
+                    json += "\"" + subservice + "\":" + metrics.toJsonString();
+                    firstSubservice = false;
+                } else {
+                    json += ",\"" + subservice + "\":" + metrics.toJsonString();
+                } // if else
+            } // for
+            
+            json += "},\"sum\":" + perServiceAggrMetrics.get(service).toJsonString() + "}";
+        } // for
+        
+        json += "},\"sum\": {\"subservs\":{";
+        boolean firstSubservice = true;
+        
+        for (String subservice : perSubserviceAggrMetrics.keySet()) {
+            Metrics metrics = perSubserviceAggrMetrics.get(subservice);
+            
+            if (firstSubservice) {
+                json += "\"" + subservice + "\":" + metrics.toJsonString();
+                firstSubservice = false;
+            } else {
+                json += ",\"" + subservice + "\":" + metrics.toJsonString();
+            } // if else
+        } // for
+
+        if (perSubserviceAggrMetrics.isEmpty()) {
+            json += "},\"sum\":{}}}";
+        } else {
+            json += "},\"sum\":" + allAggrMetrics.toJsonString() + "}}";
+        } // if else
+        
+        return json;
+    } // toJsonString
+    
+    /**
+     * Metrics class.
+     */
+    public class Metrics {
+        
+        private long incomingTransactions;
+        private long incomingTransactionRequestSize;
+        private long incomingTransactionResponseSize;
+        private long incomingTransactionErrors;
+        private long outgoingTransactions; // Hidden metric just for service time compautations
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1383/commits/0d04449570086c47f147d87af3fa804bf604b237